src/components/views/dialogs/DMInviteDialog.js

```diff
@@ -31,18 +31,46 @@ import {getHttpUriForMxc} from "matrix-js-sdk/lib/content-repo";
 const INITIAL_ROOMS_SHOWN = 3; // Number of rooms to show at first
 const INCREMENT_ROOMS_SHOWN = 5; // Number of rooms to add when 'show more' is clicked
 
-class DirectoryMember {
+// This is the interface that is expected by various components in this file. It is a bit
+// awkward because it also matches the RoomMember class from the js-sdk with some extra support
+// for 3PIDs/email addresses.
+//
+// Dev note: In order to allow us to compile the app correctly, this needs to be a class
+// even though FlowJS supports interfaces. It just means that we "extend" rather than "implement"
```

Comment:
Is this Flow the type checker? (I don't think it's called "FlowJS"...?) Assuming yes, I didn't realise we were even using that now...? 😅

Reply:
yes, Flow, not FlowJS. Everything is named badly.

We have had Flow in our project since pretty much inception, though have historically been really bad at using it. We're migrating to TypeScript, but this component is before that time :(